src/toil_scripts/tools/aligners.py

```diff
@@ -56,3 +56,79 @@ def run_star(job, cores, r1_id, r2_id, star_index_url):
     transcriptome_id = job.fileStore.writeGlobalFile(os.path.join(work_dir, 'rnaAligned.toTranscriptome.out.bam'))
     sorted_id = job.fileStore.writeGlobalFile(os.path.join(work_dir, 'rnaAligned.sortedByCoord.out.bam'))
     return transcriptome_id, sorted_id
+
+
+def run_bwakit(job, threads, r1, r2, ref, fai, amb, ann, bwt, pac, sa, uuid, library, platform, program_unit,
+               alt=None, rg_line=None, sort=True, trim=False):
+    """
+    Runs BWA-Kit to align a fastq file or fastq pair into a BAM file
+
+    :param JobFunctionWrappingJob job: Passed by Toil automatically
+    :param int threads: Number of threads to use
+    :param str r1: FileStoreID of first fastq
+    :param str r2: FileStoreID of second fastq (None if not using paired data)
+    :param str ref: FileStoreID of reference
+    :param str fai: FileStoreID of fai
+    :param str amb: FileStoreID of amb
+    :param str ann: FileStoreID of ann
+    :param str bwt: FileStoreID of bwt
+    :param str pac: FileStoreID of pac
+    :param str sa: FileStoreID of sa
+    :param str uuid: UUID of sample (for readgroup)
+    :param str library: library of sample (for readgroup)
+    :param str platform: platform of sample (for readgroup)
+    :param str program_unit: program unit for sample (for readgroup)
+    :param str alt: FileStoreID for alt sequence
+    :param str rg_line: rg line. Use the format "@RG\\tID:foo\\tLB:bar" ...
+    :param bool sort: If True, sorts the BAM
+    :param bool trim: If True, performs adapter trimming
+    :return: FileStoreID of BAM
+    :rtype: str
+    """
+    work_dir = job.fileStore.getLocalTempDir()
+    file_names = ['r1.fq.gz', 'ref.fa.fai', 'ref.fa', 'ref.fa.amb', 'ref.fa.ann',
+                  'ref.fa.bwt', 'ref.fa.pac', 'ref.fa.sa']
+    ids = [r1, ref, fai, amb, ann, bwt, pac, sa]
+    if r2:
+        file_names.insert(1, 'r2.fq.gz')
+        ids.insert(1, r2)
+    if alt:
+        file_names.append('ref.fa.alt')
+        ids.append(alt)
+    for fileStoreID, name in zip(ids, file_names):
+        job.fileStore.readGlobalFile(fileStoreID, os.path.join(work_dir, name))
+    # Read groups
+    if rg_line:
+        rg = rg_line
+    else:
+        rg = "@RG\\tID:{0}".format(uuid)  # '\' character is escaped so bwakit gets individual '\' and 't' characters
+        for tag, info in zip(['LB', 'PL', 'PU', 'SM'], [library, platform, program_unit, uuid]):
+            rg += '\\t{0}:{1}'.format(tag, info)
+    # BWA Options
+    opt_args = []
+    if sort:
+        opt_args.append('-s')
+    if trim:
+        opt_args.append('-a')
+    # Call: bwakit
+    parameters = (['-t', str(threads),
+                   '-R', rg] +
+                  opt_args +
+                  ['-o', '/data/aligned',
+                   '/data/ref.fa',
+                   '/data/r1.fq.gz'])
```

Comment:
I might've misunderstood this one — the `parameter.append` code is directly below this line now, but you could also mean move it so there's only one call to `if getattr(config, 'r2', None):` ?